shared/data/src/main/java/com/ivy/data/remote/responses/ExchangeRatesResponse.kt

```diff
@@ -1,11 +1,62 @@
 package com.ivy.data.remote.responses
 
 import androidx.annotation.Keep
+import arrow.core.Option.Companion.fromNullable
+import arrow.core.getOrElse
+import arrow.core.toOption
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.JsonTransformingSerializer
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlinx.serialization.json.jsonPrimitive
 
-@Serializable
+@Serializable(with = ExchangeRatesResponseDeserializer::class)
 @Keep
 data class ExchangeRatesResponse(
+    @SerialName("date")
```

Comment:
I think this is the real fix but why do we need a custom serializer? It's again complex 😂 Before when I reworked it to Ktor it worked w/o a custom serializer. Just `@SerialName`

Reply:
we need a custom serializer because the JSON is dynamic and there is no easy way to handle it kotlinx.serialization 
if it was jackson I coud have used @JsonAnySetter instead and fixed it up 

Dynamic Json e.g --> if Currency is USD -->
{
    "date": "2024-03-24",
    "usd": {
        "$myro": 4.62721175,
        "$wen": 3744.01152636,
        .....
        }
}

If currency is EUR  -->
{
    "date": "2024-03-24",
    "eur": {
        "$myro": 4.62721175,
        "$wen": 3744.01152636,
        .....
        }
}

We are trying to get it in 

{
date: "..."
rates: "..."
}
so rates as a json field will never be available

Reply:
Can't we just rename "rates" to "eur" in the SerialName and call it a day? We don't need that extra complexity. It creates new codepaths that can fail with no added benefit

Reply:
We don't neeb dynamic base currency. Fetch everything in EUR and then recompute to user's base currency. This computation is almost instant

Reply:
If we were only fetching eur that would be fine but lets say we fetch USD 
instead of "eur" you will get "usd" in json response

Reply:
Yes, I agree but let's fetch only EUR for the sake of simplicity and API stability. I bet our API doesn't support RSD as a base